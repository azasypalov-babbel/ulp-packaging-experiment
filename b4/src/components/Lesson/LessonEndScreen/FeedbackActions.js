import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../lib/cxHelper';
import * as features from '../../../lib/features';
import LoyButton from '../../Legacy/LoyButton';

const FeedbackActions = ({
  purgeableItemsCount,
  onCorrectErrorsButtonClick,
  onReturnHomeButtonClick,
  isUnlocked,
  onAccessContentButtonClick,
  accessLessonContentText,
  accessLessonContentButton,
  correctErrorsButton,
  returnHomeButton
}) => {
  const showCorrectErrors = purgeableItemsCount > 0 && !features.isWebview();

  const renderContentLocked = () => (
    <>
      <p>{accessLessonContentText}</p>
      <LoyButton
        primary
        listenToKey="Enter"
        data-selector="access-content-button"
        onClick={onAccessContentButtonClick}
      >
        {accessLessonContentButton}
      </LoyButton>
    </>
  );

  const renderContentUnlocked = () => (
    <>
      { showCorrectErrors && (
        <LoyButton
          primary
          data-selector="correct-errors-button"
          listenToKey="Enter"
          onClick={onCorrectErrorsButtonClick}>
          {correctErrorsButton}
        </LoyButton>
      )}
      <LoyButton
        primary={!showCorrectErrors}
        listenToKey={!showCorrectErrors ? 'Enter' : undefined}
        data-selector="return-home-button"
        onClick={onReturnHomeButtonClick}>
        {returnHomeButton}
      </LoyButton>
    </>
  );

  return (
    <div className={cx('lesson-end-screen-feedback-actions')}>
      { isUnlocked ?
        renderContentUnlocked() : renderContentLocked()
      }
    </div>
  );
};

FeedbackActions.propTypes = {
  purgeableItemsCount: PropTypes.number,
  onCorrectErrorsButtonClick: PropTypes.func,
  onReturnHomeButtonClick: PropTypes.func,
  isUnlocked: PropTypes.bool.isRequired,
  onAccessContentButtonClick: PropTypes.func,
  accessLessonContentText: PropTypes.string.isRequired,
  accessLessonContentButton: PropTypes.string.isRequired,
  correctErrorsButton: PropTypes.string.isRequired,
  returnHomeButton: PropTypes.string.isRequired
};

export default FeedbackActions;
