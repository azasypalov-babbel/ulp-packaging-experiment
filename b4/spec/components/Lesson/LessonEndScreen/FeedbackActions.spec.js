import React from 'react';
import FeedbackActions from '../../../../src/components/Lesson/LessonEndScreen/FeedbackActions';
import { shallow } from 'enzyme';

describe('<FeedbackActions />', () => {
  describe('Content Unlocked', () => {
    const defaultProps = {
      isUnlocked: true,
      onCorrectErrorsButtonClick: jest.fn(),
      onReturnHomeButtonClick: jest.fn(),
      accessLessonContentText: '',
      accessLessonContentButton: '',
      correctErrorsButton: 'Correct my mistakes (4)',
      returnHomeButton: 'Continue learning'
    };

    test('it renders when there are purgable items', () => {
      const props = Object.assign({}, defaultProps, {
        purgeableItemsCount: 4
      });
      const component = shallow(
        <FeedbackActions {...props} />
      );

      expect(component).toMatchSnapshot();
    });

    test('it renders when there are no purgable items', () => {
      const props = Object.assign({}, defaultProps, {
        purgeableItemsCount: 0
      });
      const component = shallow(
        <FeedbackActions {...props} />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('Content Locked', () => {
    const props = {
      isUnlocked: false,
      onAccessContentButtonClick: jest.fn(),
      accessLessonContentText: 'Keep it up! Get access to all lessons and topics.',
      accessLessonContentButton: 'Access all French content',
      correctErrorsButton: '',
      returnHomeButton: ''
    };

    test('it renders', () => {
      const component = shallow(
        <FeedbackActions {...props} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
