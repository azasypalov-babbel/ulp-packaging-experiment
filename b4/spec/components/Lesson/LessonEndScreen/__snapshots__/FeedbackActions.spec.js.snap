// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackActions /> Content Locked it renders 1`] = `
<div
  className="loy-lesson-end-screen-feedback-actions"
>
  <p>
    Keep it up! Get access to all lessons and topics.
  </p>
  <WithKeypress(LoyButton)
    data-selector="access-content-button"
    listenToKey="Enter"
    onClick={[MockFunction]}
    primary={true}
  >
    Access all French content
  </WithKeypress(LoyButton)>
</div>
`;

exports[`<FeedbackActions /> Content Unlocked it renders when there are no purgable items 1`] = `
<div
  className="loy-lesson-end-screen-feedback-actions"
>
  <WithKeypress(LoyButton)
    data-selector="return-home-button"
    listenToKey="Enter"
    onClick={[MockFunction]}
    primary={true}
  >
    Continue learning
  </WithKeypress(LoyButton)>
</div>
`;

exports[`<FeedbackActions /> Content Unlocked it renders when there are purgable items 1`] = `
<div
  className="loy-lesson-end-screen-feedback-actions"
>
  <WithKeypress(LoyButton)
    data-selector="correct-errors-button"
    listenToKey="Enter"
    onClick={[MockFunction]}
    primary={true}
  >
    Correct my mistakes (4)
  </WithKeypress(LoyButton)>
  <WithKeypress(LoyButton)
    data-selector="return-home-button"
    onClick={[MockFunction]}
    primary={false}
  >
    Continue learning
  </WithKeypress(LoyButton)>
</div>
`;
